File: VB/E4931/XRHyperlink.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrinting.BrickExporters
Imports DevExpress.XtraPrinting.Native
Imports DevExpress.XtraPrinting.Export
Imports DevExpress.XtraReports.UI

Namespace E4931
	Friend Class HyperlinkBrickExporter
		Inherits LabelBrickExporter
		Protected ReadOnly Property HyperlinkBrick() As HyperlinkBrick
			Get
				Return TryCast(Brick, HyperlinkBrick)
			End Get
		End Property

		Protected Overrides Sub FillHtmlTableCellCore(ByVal exportProvider As IHtmlExportProvider)
			exportProvider.CurrentCell.InnerHtml = String.Format("<a href='{1}' target='{2}'>{0}</a>", HttpUtility.HtmlEncode(HyperlinkBrick.Text), HyperlinkBrick.Url, HyperlinkBrick.Target)
		End Sub
	End Class

	<BrickExporter(GetType(HyperlinkBrickExporter))> _
	Public Class HyperlinkBrick
		Inherits LabelBrick
		Public Shared HyperlinkBrickTypeName As String = "Hyperlink"

		Shared Sub New()
			BrickFactory.RegisterFactory(HyperlinkBrickTypeName, New DefaultBrickFactory(Of HyperlinkBrick)())
		End Sub

		Public Sub New()
			MyBase.New()
		End Sub

		Public Sub New(ByVal owner As IBrickOwner)
			MyBase.New(owner)
		End Sub

		Public Overrides ReadOnly Property BrickType() As String
			Get
				Return HyperlinkBrickTypeName
			End Get
		End Property
	End Class

	Public Class XRHyperlink
		Inherits XRLabel
		Protected Overrides Function CreateBrick(ByVal childrenBricks() As VisualBrick) As VisualBrick
			Return New HyperlinkBrick(Me)
		End Function
	End Class
End Namespace

File: CS/E4931/XRHyperlink.cs
using System;
using System.Web;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrinting.BrickExporters;
using DevExpress.XtraPrinting.Native;
using DevExpress.XtraPrinting.Export;
using DevExpress.XtraReports.UI;

namespace E4931 {
    class HyperlinkBrickExporter : LabelBrickExporter {
        protected HyperlinkBrick HyperlinkBrick {
            get {
                return Brick as HyperlinkBrick;
            }
        }

        protected override void FillHtmlTableCellCore(IHtmlExportProvider exportProvider) {
            exportProvider.CurrentCell.InnerHtml = String.Format("<a href='{1}' target='{2}'>{0}</a>",
                HttpUtility.HtmlEncode(HyperlinkBrick.Text),
                HyperlinkBrick.Url,
                HyperlinkBrick.Target);
        }
    }

    [BrickExporter(typeof(HyperlinkBrickExporter))]
    public class HyperlinkBrick : LabelBrick {
        public static string HyperlinkBrickTypeName = "Hyperlink";

        static HyperlinkBrick() {
            BrickFactory.RegisterFactory(HyperlinkBrickTypeName, new DefaultBrickFactory<HyperlinkBrick>());
        }

        public HyperlinkBrick()
            : base() {
        }

        public HyperlinkBrick(IBrickOwner owner)
            : base(owner) {
        }

        public override string BrickType {
            get {
                return HyperlinkBrickTypeName;
            }
        }
    }

    public class XRHyperlink : XRLabel {
        protected override VisualBrick CreateBrick(VisualBrick[] childrenBricks) {
            return new HyperlinkBrick(this);
        }
    }
}